src/porepy/numerics/mixed_dim/dof_manager.py

```diff
@@ -198,42 +196,115 @@ def distribute_variable(
+                the solution of a linear system assembled with the assembler.
+            variable_names (list of str, optional): Names of the variable to be
+                distributed. If not provided, all variables found in block_dof
+                will be distributed
+            additive (bool, optional): If True, the variables are added to the current
+                state, instead of overwrite the existing value.
+
+        """
+        if variable_names is None:
+            variable_names = []
+            for pair in self.block_dof.keys():
+                variable_names.append(pair[1])
+
+        # Make variable_names unique
+        variable_names = list(set(variable_names))
+
+        # Determine restricted dof set for provided variables
+        var_num_dofs = [self.num_dofs(var=v) for v in variable_names]
+        dof = np.cumsum(np.append(0, np.asarray(var_num_dofs)))
 
-        # Return the restricted dofs
-        return projection * global_dofs
+        for var_name in set(variable_names):
+            vi = variable_names.index(var_name)
+            for pair, bi in self.block_dof.items():
+                g = pair[0]
+                name = pair[1]
+                if name != var_name:
+                    continue
+                if isinstance(g, tuple):
+                    # This is really an edge
+                    data = self.gb.edge_props(g)
+                else:
+                    data = self.gb.node_props(g)
+
+                if not(iterate):
+                    if pp.STATE in data.keys():
+                        vals = values[dof[vi] : dof[vi + 1]]
+                        if additive:
+                            vals += data[pp.STATE][var_name]
+
+                        data[pp.STATE][var_name] = vals
+                    else:
+                        # If no values exist, there is nothing to add to
+                        data[pp.STATE] = {var_name: values[dof[vi] : dof[vi + 1]]}
+                else:
+                    if pp.STATE in data.keys():
+                        vals = values[dof[vi] : dof[vi + 1]]
+                        if pp.ITERATE in data[pp.STATE].keys():
+                            if additive:
+                                vals += data[pp.STATE][pp.ITERATE][var_name]
+
+                            data[pp.STATE][pp.ITERATE][var_name] = vals
+                        else:
+                            data[pp.STATE][pp.ITERATE] = {}
+                            data[pp.STATE][pp.ITERATE][var_name] = vals
+
+                    else:
+                        # If no values exist, there is nothing to add to
+                        data[pp.STATE] = {}
+                        data[pp.STATE][pp.ITERATE] = {var_name: values[dof[vi] : dof[vi + 1]]}
+
+    def transform_dofs(self, dofs: np.ndarray, var: Optional[list]) -> np.ndarray:
+        """Transforms dofs associated to full list of dofs to a restricted list of dofs.
+
```

Comment:
Please describe the output (standard documentation), and how it relates to the input.

Reply:
.. that is, I what does the method really do?